src/main/java/com/google/devtools/build/lib/runtime/QueryRuntimeHelper.java

```diff
@@ -63,8 +68,20 @@ class StdoutQueryRuntimeHelperFactory implements Factory {
     private StdoutQueryRuntimeHelperFactory() {}
 
     @Override
-    public QueryRuntimeHelper create(CommandEnvironment env) {
-      return createInternal(env.getReporter().getOutErr().getOutputStream());
+    public QueryRuntimeHelper create(CommandEnvironment env, CommonQueryOptions options)
+        throws QueryRuntimeHelperException {
+      PathFragment outputFile = options.outputFile;
+
+      if (outputFile == null) {
+        return createInternal(env.getReporter().getOutErr().getOutputStream());
+      }
+      Path fullPath = env.getWorkspace().getRelative(outputFile);
```

Comment:
Should this be `CommandEnvironment#getWorkspace` or `CommandEnvironment#getWorkingDirectory`? The latter should let you store the file beneath the workspace root?